src/main/webapp/app/shared/components/clone-repo-button/clone-repo-button.component.ts

```diff
@@ -99,6 +102,43 @@ export class CloneRepoButtonComponent implements OnInit, OnChanges {
         }
     }
 
+    /**
+     * Called when the user clicks on the clone repo button. Sometimes the VCS access token is not present in the user
+     * object, so the resulting repo URL is wrong. This method tries to retrieve the token from the server by requesting
+     * the user object again.
+     * This case only happens when a user is created before VCS access tokens are enabled. For those users, a token is
+     * created the first time they start participating in a programming exercise. In that case, however, they do not
+     * have an access token yet, because during login, there was no token available.
+     *
+     * In case the token could not be retrieved, this method is not called again and the student
+     * is prompted to reload the page. Otherwise, the error would be displayed every time the button is clicked,
+     * including the clicks that make the popover go away (every second click).
+     */
+    onClick() {
+        if (this.versionControlAccessTokenRequired && !this.user.vcsAccessToken && !this.unableToLoadVCSAccessToken) {
```

Comment:
It is best practice not to negate boolean variables. It would be nice if you could introduce ableToLoasVCSAccessToken (instead of unableTo...). This makes the code easier to read.